### DISCUSSION TITLE
How to account for transform : scale(n) to preserve the width of the bars?

### DISCUSSION THREAD

**Discussion by dmytro-kuz** [katspaugh/wavesurfer.js#3195] [Q&A] [answered]
Hi, I have a working example at https://codepen.io/Dmytro-Kuz/pen/zYydOKp where the waveform container is set to transform: scale(n). In this case, the bar width is not preserved and is also scaled.
Is there a way to retain the original width value for barWidth? And if I perform additional calculations to set a new value for barWidth every time the scaleFactor changes, do I need to destroy the waveform and create a new instance to set the new barWidth value?

Also, if you look closely at the scaled bars, you can see that the width of the bars is slightly different... Any ideas how to solve my issue?

**Comment by katspaugh** [comment, marked as answer]
You don’t have to re-create it, just set the bar width like this:
`ws.setOptions({ barWidth: 10 })`
